site/app/libraries/DateUtils.php

```diff
@@ -77,4 +77,47 @@ public static function validateTimestamp($timestamp) {
         }
         return true;
     }
+
+
+    /**
+     * Asserts that the provided date is a \DateTime object and converts it to one
+     *  if its a string, returning any error in parsing.
+     *
+     * @param $date \DateTime|string A reference to the date object to assert.  Set to null if failed.
+     * @param $time_zone \DateTimeZone The timezone to parse with
+     * @return null|string The error message or null
+     */
+    public static function assertDate(&$date, \DateTimeZone $time_zone) {
```

Comment:
So couple things:
1. a function prefixed with "assert" I'd expect to either return true/false or throw an "AssertionError" (looking at the assert function in PHP or python) and that's it
2. I dislike the paradigm of having a function that takes in a variable, mutates it, and then returns null (it was successfully mutated) or a string (to indicate an error). To me, a function should either a) mutate the variables passed by reference or b) mutate variable passed by value and return the changed value (and I drastically prefer b). In either case, given this is OOP, throw an exception if we hit an invalid value for that mutation.
3. If your function definition contains an "and' to describe two discrete things that it's doing and that neither feed from the other, this should be two different functions.